### ISSUE TITLE
Consider UX on meltano discover with Hub as SSOT

### ISSUE THREAD

**Issue by MeltyBot** [meltano/meltano#3430]
Migrated from GitLab: https://gitlab.com/meltano/meltano/-/issues/3514

Originally created by @tayloramurphy on 2022-05-19 21:40:07

---

https://docs.meltano.com/reference/command-line-interface#discover is an okay experience when there are < 25 plugins available for a particular type. We're now going to have over 300 taps listed if a user does `meltano discover all` or `meltano discover extractors`.

Is there something we can do that's quick for ~"v2.0" to make it better? I'm thinking:

* Limit the number of results returned to ~30 and have a link to the Hub to learn more. 
* Update each plugin to actually link to the Hub, or at least display the link. 

I'm not sure of the long term utility of `meltano discover` in a world where we have the Hub, at least in its current state.

@edgarrmondragon @aaronsteers

**Comment by MeltyBot** [comment]
[View 8 previous comments from the original issue on GitLab](https://gitlab.com/meltano/meltano/-/issues/3514)

**Comment by edgarrmondragon** [comment]
I thought of this as having a CLI to the Hub search, where users could see a subset of the plugin information displayed on the Hub, without leaving the terminal. I'd also be okay with removing it.

**Comment by pnadolny13** [comment]
@WillDaSilva last month at least 155 projects ran discover and last year 3k+ projects used it. Zooming into last month, most of them ran it only once or twice but theres some that run it hundreds of times. I would guess there are some "powered by meltano" apps that use the discover command as a way of listing all available plugins. I wonder if Matatika does this?

Its also possible that many of these users are running discover while following our docs and guides.
- https://docs.meltano.com/getting-started/#add-an-extractor-to-pull-data-from-a-source
- https://docs.meltano.com/guide/transformation#install-dbt
- https://docs.meltano.com/guide/mappers#standalone-mapper-plugins
- etc.

Personally I dont get any value out of that command. I agree with Edgar that having it be a CLI for the hub could make it more useful but I'd probably still go to the hub directly.

**Comment by tayloramurphy** [comment]
Let's bring this to office hours. Could be fun to chat about.

**Comment by WillDaSilva** [comment]
Closing in favour of:
- https://github.com/meltano/meltano/issues/7657
- https://github.com/meltano/meltano/issues/7656
